src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -271,17 +271,20 @@ public override void Write(Utf8JsonWriter writer, T value, JsonSerializerOptions
         internal override T ReadAsPropertyNameCore(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
         {
 #if NETCOREAPP
-            bool success = TryParseEnumCore(ref reader, out T value);
+            if (TryParseEnumCore(ref reader, out T value))
```

Comment:
The fix is updating the `ReadAsPropertyNameCore` method to be in line with the `Read` method implementation.